src/schema/firefox-schemas-import.js

```diff
@@ -250,12 +250,12 @@ inner.updateWithAddonsLinterData = (firefoxSchemas, ourSchemas) => {
       schemas[ourSchema.id || namespace] = {
         ...firefoxSchema,
         file,
-        schema: merge(firefoxSchema.schema, ourSchema),
+        schema: deepmergeWithComplexArrays(firefoxSchema.schema, ourSchema),
```

Comment:
Let's add an inline comment here and at line 258 to briefly explain that here we are using a different algorithm to merge the two objects because we are actually "patching" the imported schema with some "addons-linter"-specific tweaks.

Reply:
Done in dc639e2c